src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/AssessmentService.kt

```diff
@@ -102,6 +104,13 @@ class AssessmentService(
   @Transactional
   fun optOut(prisonNumber: String, optOutRequest: OptOutRequest) {
     transitionAssessment(prisonNumber, OptOut, optOutRequest.agent)
+
+    val assessmentEntity = getCurrentAssessment(prisonNumber).assessmentEntity
```

Comment:
I think it'd be better to extract out the transition assessment.. to avoid the two db look ups and saves.

```
... getcurrentAssessment
transition
set opt out reason
set opt out reasonn other
save..
```

Reply:
I get this is basically doing the same as from `updateAddressChecksStatus`

Reply:
Yes did copy that code have updated to your suggested approach, looks better makes sense